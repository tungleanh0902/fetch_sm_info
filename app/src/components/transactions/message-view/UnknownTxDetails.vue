<template>
  <div class="tx__content">
    <TransactionIcon :transaction-type="type" />
    <div class="tx__content__left">
      <h3>{{ caption }}</h3>
    </div>
  </div>
</template>

<script>
import TransactionIcon from "../TransactionIcon"

export default {
  name: `unknown-tx-details`,
  components: {
    TransactionIcon,
  },
  props: {
    transaction: {
      type: Object,
      required: true,
    },
    validators: {
      type: Object,
      required: true,
    },
  },
  data: () => {
    return {
      type: `Submitted`,
      caption: `Unknown`,
    }
  },
}
</script>
<style scoped>
.italic {
  font-style: italic;
}
</style>
