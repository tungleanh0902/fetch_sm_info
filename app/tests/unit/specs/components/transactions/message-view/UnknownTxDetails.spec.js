import { shallowMount } from "@vue/test-utils"
import UnknownTxDetails from "src/components/transactions/message-view/UnknownTxDetails"

describe(`UnknownTxDetails`, () => {
  let wrapper

  const tx = {
    type: "UnknownTxDetails",
    hash: "3CA728671B8078E71697B62237AD694052779F80B56880F6A6F1702F53EA3081",
    height: 308453,
    timestamp: "2020-01-10T09:02:54Z",
    memo: "",
    success: true,
    fees: [
      {
        denom: "ATOM",
        amount: "0.00045",
      },
    ],
    details: {
      blockExplorerLink:
        "https://awesome.blockexplorer.io/transactions/3CA728671B8078E71697B62237AD694052779F80B56880F6A6F1702F53EA3081",
    },
  }

  it(`renders a unstake transaction message`, () => {
    wrapper = shallowMount(UnknownTxDetails, {
      propsData: {
        transaction: tx,
        validators: {},
      },
      stubs: [`router-link`],
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
