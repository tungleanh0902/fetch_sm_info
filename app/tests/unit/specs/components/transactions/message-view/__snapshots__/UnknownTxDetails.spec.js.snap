// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnknownTxDetails renders a unstake transaction message 1`] = `
<div
  class="tx__content"
>
  <transactionicon-stub
    transactiontype="Submitted"
  />
   
  <div
    class="tx__content__left"
  >
    <h3>
      Unknown
    </h3>
  </div>
</div>
`;
